crypto/x509/x509_lu.c

```diff
@@ -310,22 +310,23 @@ int X509_STORE_add_cert(X509_STORE *ctx, X509 *x)
         return 0;
     obj->type = X509_LU_X509;
     obj->data.x509 = x;
+    X509_OBJECT_up_ref_count(obj);
 
     CRYPTO_THREAD_write_lock(ctx->lock);
 
-    X509_OBJECT_up_ref_count(obj);
-
     if (X509_OBJECT_retrieve_match(ctx->objs, obj)) {
-        X509_OBJECT_free(obj);
         X509err(X509_F_X509_STORE_ADD_CERT,
                 X509_R_CERT_ALREADY_IN_HASH_TABLE);
         ret = 0;
     } else
-        sk_X509_OBJECT_push(ctx->objs, obj);
+        ret = sk_X509_OBJECT_push(ctx->objs, obj);
```

Comment:
The first branch of the `if` is enclosed ind `{}` and therefore this line should be too according to our coding style.